### PR TITLE
Tests for pbs version header

### PR DESCRIPTION
**Pull request by mtuchkova**
Tests for #1537

### CODE REVIEW

File: src/test/groovy/org/prebid/server/functional/model/response/amp/AmpResponse.groovy [hunk 1]
```diff
@@ -1,11 +1,7 @@
...
 
-import org.prebid.server.functional.model.response.BidderError
-import org.prebid.server.functional.model.response.Debug
-
 class AmpResponse {
 
-    Map<String, String> targeting
-    Debug debug
-    Map<String, List<BidderError>> errors
+    AmpResponseBody responseBody
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/model/response/amp/AmpResponse.groovy:5]: Bad idea. You will need headers for only a couple of tests, but you are complicating model for everything else. Better add a method to a PBS service to return raw response body and headers.

File: src/test/groovy/org/prebid/server/functional/model/response/auction/AuctionResponse.groovy [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package org.prebid.server.functional.model.response.auction
+
+import groovy.transform.ToString
+import org.prebid.server.functional.model.ResponseModel
+
+@ToString(includeNames = true, ignoreNulls = true)
+class AuctionResponse implements ResponseModel {
+
+    BidResponse responseBody
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/model/response/auction/AuctionResponse.groovy:9]: Same as with AMP. You are complicating things where you should not.

File: src/test/groovy/org/prebid/server/functional/model/response/amp/AmpResponse.groovy (hunk 1 above)
Comment (Net-burst) [reply src/test/groovy/org/prebid/server/functional/model/response/amp/AmpResponse.groovy:5]: You can leave this class as a wrapper for that service method to return, but it should not replace existing interactions. But the name will need to change.

File: src/test/groovy/org/prebid/server/functional/AmpSpec.groovy [hunk 3]
```diff
@@ -121,6 +121,25 @@ class AmpSpec extends BaseSpec {
...
+        and: "Save storedRequest into DB"
+        def storedRequest = StoredRequest.getDbStoredRequest(ampRequest, ampStoredRequest)
+        storedRequestDao.save(storedRequest)
+
+        when: "PBS processes amp request"
+        def response = defaultPbsService.sendAmpRequest(ampRequest)
+
+        then: "Response header should contain PBS version"
+        def bidderHeaders = bidder.getRecordedRequestsHeaders(ampStoredRequest.id)[0]
+        assert response.headers["x-prebid"]
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/AmpSpec.groovy:139]: You should check that actual project version from `pom.xml` is passed.

**Review by Net-burst — CHANGES_REQUESTED** [review]

**Review by And1sS — COMMENTED** [review]
Could you, please, add test that checks header on failed request?

**Review by And1sS — APPROVED** [review]
LGTM! Thanks!

**Review by Net-burst — APPROVED** [review]
LGTM

**Pull request merged**